### PR TITLE
[Method Reference: SolrClientManager]Lps 180691

### PR DESCRIPTION
**Pull request by slnn**
https://liferay.atlassian.net/browse/LPS-180691

### CODE REVIEW

**Comment by slnn** [comment]
ci:test:search-solr

**Comment by slnn** [comment]
ci:test:search

**Comment by Kvale1733** [comment]
ci:test:stable

**Comment by Kvale1733** [comment]
@slnn No regressions found.

File: modules/apps/portal-search-solr8/portal-search-solr8-impl/src/main/java/com/liferay/portal/search/solr8/internal/connection/SolrClientManager.java [hunk 1]
```diff
@@ -123,8 +112,6 @@ protected void setCloudSolrClientFactory(
 
 	@Reference(
 		cardinality = ReferenceCardinality.MULTIPLE,
-		policy = ReferencePolicy.DYNAMIC,
```
Comment (dantewang) [inline modules/apps/portal-search-solr8/portal-search-solr8-impl/src/main/java/com/liferay/portal/search/solr8/internal/connection/SolrClientManager.java:126]: @slnn Basically looks good to me, but please don't remove the DYNAMIC GREEDY from MULTIPLE references. We just want to make MANDATORY references static.

**Review by dantewang — CHANGES_REQUESTED** [review]

File: modules/apps/portal-search-solr8/portal-search-solr8-impl/src/main/java/com/liferay/portal/search/solr8/internal/connection/SolrClientManager.java (hunk 1 above)
Comment (dantewang) [reply modules/apps/portal-search-solr8/portal-search-solr8-impl/src/main/java/com/liferay/portal/search/solr8/internal/connection/SolrClientManager.java:126]: @slnn I mean in this commit.

**Comment by dantewang** [comment]
@slnn By the way, could you ask Tibor whether this needs to be recorded as a breaking change? Thanks!

File: modules/apps/portal-search-solr8/portal-search-solr8-impl/src/main/java/com/liferay/portal/search/solr8/internal/connection/SolrClientManager.java (hunk 1 above)
Comment (slnn) [reply modules/apps/portal-search-solr8/portal-search-solr8-impl/src/main/java/com/liferay/portal/search/solr8/internal/connection/SolrClientManager.java:126]: Done

**Comment by slnn** [comment]
Resend in https://github.com/liferay-core-infra/liferay-portal/pull/4913

**Comment by slnn** [comment]
> @slnn By the way, could you ask Tibor whether this needs to be recorded as a breaking change? Thanks!

Hi @dantewang 

Confirmed with Tibor, this should be a breaking change, https://liferay.atlassian.net/browse/PTR-4058

Lily

**Comment by tinatian** [comment]
See https://github.com/liferay-core-infra/liferay-portal/pull/5018

@slnn Please note I modified the commit message at https://github.com/liferay-core-infra/liferay-portal/pull/5018/commits/e40de956c7663a0514853407b9ec9edab22b28ed.

The reason you removed dynamic is not because we have bug. It is we decided not to support.

When you have bug, you should fix the issue rather than remove the feature.

Thanks.
Tina.

**Pull request closed without merging**